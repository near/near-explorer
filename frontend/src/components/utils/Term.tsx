import React from "react";

import Modal from "./Modal";

interface Props {
  title: string;
}

interface State {
  modalInfo?: string;
}

export default class extends React.Component<Props, State> {
  state: State = {};

  changeModalHandler = (name: string) => {
    this.setState({ modalInfo: name });
  };

  ModalCloseHander = () => {
    this.setState({ modalInfo: undefined });
  };

  render() {
    const { title } = this.props;
    return (
      <>
        {title}
        <div className="term-helper">
          <img
            src="/static/images/icon-info.svg"
            className="info"
            onClick={() => this.changeModalHandler(title)}
          />
          {this.state.modalInfo && (
            <Modal modalClosed={this.ModalCloseHander}>
              <h4>{title}</h4>
              {this.props.children}
            </Modal>
          )}
          <style jsx global>{`
            .term-helper {
              display: inline-block;
              width: 14px;
              height: 14px;
              text-transform: none;
            }

            .term-helper .info {
              vertical-align: text-bottom;
              margin-left: 5px;
              width: 16px;
              cursor: pointer;
            }
          `}</style>
        </div>
      </>
    );
  }
}
