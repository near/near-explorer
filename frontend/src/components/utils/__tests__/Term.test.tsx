import renderer from "react-test-renderer";

import Term from "../Term";

describe("<CardCell />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <Term title={"Nodes Online"}>
          {
            "The number of validating nodes / the total number of online nodes. "
          }
          <a
            href={
              "https://docs.nearprotocol.com/docs/roles/integrator/faq#validators"
            }
          >
            docs
          </a>
        </Term>
      )
    ).toMatchSnapshot();
  });
});
