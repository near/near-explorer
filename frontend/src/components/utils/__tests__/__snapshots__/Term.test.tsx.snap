// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCell /> renders 1`] = `
Array [
  "Nodes Online",
  <div
    className="jsx-1104606862 term-helper"
  >
    <img
      className="jsx-1104606862 info"
      onClick={[Function]}
      src="/static/images/icon-info.svg"
    />
  </div>,
]
`;
